packages/gatsby-plugin-theme-ui/gatsby-node.js

```diff
@@ -0,0 +1,58 @@
+exports.onPreInit = (__, options) => {
+  let { themePreset } = options
+
+  if (typeof themePreset === 'string') {
+    try {
+      options.themePreset = require(themePreset)
```

Comment:
NB: Is this mutating the `options` object? Might be nice to define a new variable in this module's scope

Reply:
If this doesn't mutate then we can't re-use themePreset to work for both JSON and a string because it won't resolve to JSON at this stage, making it incompatible with the schema customization.